### PR TITLE
LoRa adapter: Check for command endpoint before creating consumer

### PR DESCRIPTION
**Pull request by calohmn**
The registration of a command consumer is skipped now during handling of a message from a LoRa network gateway if there is no command endpoint defined in the gateway device registration data. 
Handling of command messages for such a gateway would fail anyway because of the missing command endpoint.

It is also made sure that the LoRa adapter "process message" span gets finished before the root span.

This relates to #2704, not creating a consumer in the first place if the gateway device isn't configured accordingly.

### CODE REVIEW

**Comment by calohmn** [comment]
@BobClaerhout Could you have a look here?

A scenario that wouldn't anymore with this change would involve a gateway device being first provisioned *without* a command endpoint and the gateway forwarding telemetry messages already.
When at a later point the "command endpoint" information gets set for the gateway device, and no telemetry messages get sent anymore, incoming command messages are not processed.
FMPOV, such a scenario isn't really realistic. WDYT?

**Comment by BobClaerhout** [comment]
Does this mean that every device to which you want to send a command to, needs to send a telemetry/event message before being able to receive a command? Or is 1 message from 1 device via that gateway/LNS enough?

**Comment by calohmn** [comment]
> Or is 1 message from 1 device via that gateway/LNS enough?

Yes, nothing has changed in that regard.
What is added here is the condition that when the telemetry/event message sent via the gateway/LNS is handled, the gateway/LNS needs to have a non-empty "command endpoint" set in its registration data, otherwise the ability to receive commands won't be activated.

**Comment by BobClaerhout** [comment]
Ok, in that case I do not have any objections. LGTM

File: adapters/lora-vertx-base/src/main/java/org/eclipse/hono/adapter/lora/LoraProtocolAdapter.java [hunk 1]
```diff
@@ -235,78 +238,106 @@ protected void customizeDownstreamMessageProperties(final Map<String, Object> pr
...
+        final String tenantId = gatewayDevice.getTenantId();
+        final String gatewayId = gatewayDevice.getDeviceId();
+        final SubscriptionKey key = new SubscriptionKey(tenantId, gatewayId);
+        if (commandSubscriptions.containsKey(key)) {
+            return;
         }
-        currentSpan.finish();
+        // use FOLLOWS_FROM span since this operation is decoupled from the rest of the request handling
+        final Span currentSpan = TracingHelper.buildFollowsFromSpan(tracer, context, "create command consumer")
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
```
Comment (sophokles73) [inline adapters/lora-vertx-base/src/main/java/org/eclipse/hono/adapter/lora/LoraProtocolAdapter.java:314]: IMHO this span's kind should rather be *client* as we are now creating the consumer and invoke the Command Router service ...

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**